Example/ConsentViewController_ExampleTests/MessageWebViewControllerSpec.swift

```diff
@@ -110,31 +110,34 @@ class MessageWebViewControllerSpec: QuickSpec, GDPRConsentDelegate, WKNavigation
+                            messageWebViewController.webview = webviewMock
+                            let message = MessageMock([
+                                "name": "onAction",
+                                "body": ["type": 12, "id": "id", "payload": ["pm_url": "https://notice.sp-prod.net/privacy-manager/index.html?message_id=122058"]]
+                            ])
 
-                        messageWebViewController.userContentController(userContentController, didReceive: message)
-                        let expectedPMURL = "https://cdn.privacy-mgmt.com/privacy-manager/index.html?site_id=1&consentUUID=uuid&message_id=122058"
-                        expect(webviewMock.loadCalledWith.url?.absoluteString).to(equal(expectedPMURL))
+                            messageWebViewController.userContentController(userContentController, didReceive: message)
+                            expect(messageWebViewController.pmId).to(equal("122058"))
+                        }
                     }
 
-                    it("calls the onAction on the consent delegate with ShowPrivacyManager with wrong PM URL") {
-                        let webviewMock = WebViewMock()
-                        messageWebViewController.connectivityManager = ConnectivityMock(connected: true)
-                        messageWebViewController.webview = webviewMock
-                        let message = MessageMock([
-                            "name": "onAction",
-                            "body": ["type": 12, "id": "id", "payload": ["pm_url": "pm_url"]]
-                        ])
-                        messageWebViewController.userContentController(userContentController, didReceive: message)
-                        let expectedPMURL = "https://cdn.privacy-mgmt.com/privacy-manager/index.html?site_id=1&consentUUID=uuid&message_id=1234"
-                        expect(webviewMock.loadCalledWith.url?.absoluteString).to(equal(expectedPMURL))
+                    context("and the pm_url attribute is invalid") {
+                        it("uses its own pmId as message_id") {
+                            let webviewMock = WebViewMock()
+                            messageWebViewController.connectivityManager = ConnectivityMock(connected: true)
+                            messageWebViewController.webview = webviewMock
+                            let message = MessageMock([
+                                "name": "onAction",
+                                "body": ["type": 12, "id": "id", "payload": ["pm_url": "pm_url"]]
+                            ])
+                            messageWebViewController.userContentController(userContentController, didReceive: message)
+                            let expectedPMURL = "https://cdn.privacy-mgmt.com/privacy-manager/index.html?site_id=1&consentUUID=uuid&message_id=1234"
+                            expect(webviewMock.loadCalledWith.url?.absoluteString).to(equal(expectedPMURL))
```

Comment:
Here, instead of checking the full URL, we can simply check the `message_id` has the value `1234`. The result is the same, but it illustrates a little bit more explicitly what the functionality is doing (changing the `message_id`).

Reply:
@andresilveirah updated the expectation.